<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from irb4400l_30_243.xacro          | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="abb_irb4400l_30_243" xmlns:xacro="http://wiki.ros.org/xacro">
  <!-- Conversion were obtained from http://www.e-paint.co.uk/Lab_values.asp
       unless otherwise stated. -->
  <!-- link list 

  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control">
    </plugin>
  </gazebo>-->
    

  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/base_link.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.7372549 0.3490196 0.1607843 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/base_link.stl"/>
      </geometry>
      <material name="">
        <color rgba="1 1 0 1"/>
      </material>
    </collision>
    <inertial>
      <mass value="1.607302"/>
      <inertia ixx="271.437744e-5" ixy="-0.003612e-5" ixz="-11.7298e-5" iyy="-0.003612e-5" iyz="509.365997e-5" izz="-11.729802e-5"/>
    </inertial>


  </link>
  <link name="link_1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/link_1.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.7372549 0.3490196 0.1607843 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/link_1.stl"/>
      </geometry>
      <material name="">
        <color rgba="1 1 0 1"/>
      </material>
    </collision>
    <inertial>
      <mass value="5.911244"/>
      <inertia ixx="2504.729736e-5" ixy="-60.221985e-5" ixz="-150.717072e-5" iyy="-60.221985e-5" iyz="2608.912842e-5" izz="-150.717072e-5"/>
    </inertial>
  </link>
  <link name="link_2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/link_2.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.7372549 0.3490196 0.1607843 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/link_2.stl"/>
      </geometry>
      <material name="">
        <color rgba="1 1 0 1"/>
      </material>
    </collision>
    <inertial>
      <mass value="1.052036"/>
      <inertia ixx="505.5751e-5" ixy="0.188865e-5" ixz="0.961193e-5" iyy="0.188865e-5" iyz="515.97e-5" izz="0.961193e-5"/>
    </inertial>
  </link>
  <link name="link_3">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/link_3.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.7372549 0.3490196 0.1607843 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/link_3.stl"/>
      </geometry>
      <material name="">
        <color rgba="1 1 0 1"/>
      </material>
    </collision>
    <inertial>
      <mass value="1.053814"/>
      <inertia ixx="94.757133e-5" ixy="-5.372609e-5" ixz="6.805163e-5" iyy="-5.372609e-5" iyz="238.8476e-5" izz="6.806e-5"/>
    </inertial>
  </link>


  <link name="link_4n">
    <visual>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/link_4n.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="">
        <color rgba="0.7372549 0.3490196 0.1607843 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/link_4n.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.108330"/>
      <inertia ixx="78370.102e-10" ixy="0.0" ixz="0.0" iyy="0.0" iyz="3386594.25e-10" izz="0.0"/>
    </inertial>
  </link>
 


  <link name="link_4">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/link_4.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.7372549 0.3490196 0.1607843 1"/>
      </material>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/link_4.stl"/>
      </geometry>
      <material name="">
        <color rgba="1 1 0 1"/>
      </material>
    </collision>

    <inertial>
      <mass value="0.833961"/>
      <inertia ixx="31.5857e-5" ixy="-2.7e-5" ixz="-0.025e-5" iyy="-2.706e-5" iyz="584.833e-5" izz="-0.025e-5"/>
    </inertial>
  </link>


  <link name="link_5">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/link_5.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.7372549 0.3490196 0.1607843 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/link_5.stl"/>
      </geometry>
      <material name="">
        <color rgba="1 1 0 1"/>
      </material>
    </collision>
    <inertial>
      <mass value="0.026219"/>
      <inertia ixx="26613.5e-10" ixy="0.0" ixz="0.0" iyy="0.0" iyz="57679.074e-10" izz="0.0"/>
    </inertial>
  </link>
  <link name="link_6">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/visual/link_6.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.1882353 0.1960784 0.2039216 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://abb_irb4400_support/meshes/irb4400l_30_243/collision/link_6.stl"/>
      </geometry>
      <material name="">
        <color rgba="1 1 0 1"/>
      </material>
    </collision>
    <inertial>
      <mass value="0.007094"/>
      <inertia ixx="3490.3e-10" ixy="-5.7e-10" ixz="-92.9e-10" iyy="-5.7e-10" iyz="18374.93e-10" izz="-92.9e-10"/>
    </inertial>
  </link>
  <link name="tool0"/>
  <!-- end of link list -->
  <!-- joint list -->
  <joint name="joint_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="link_1"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-2.87979" upper="2.87979" velocity="2.618"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <joint name="joint_2" type="revolute">
    <origin rpy="0 0 0" xyz="0.2 0 0.68"/>
    <parent link="link_1"/>
    <child link="link_2"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1.2217" upper="1.658" velocity="2.0944"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <joint name="joint_3" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.89"/>
    <parent link="link_2"/>
    <child link="link_3"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1.0472" upper="1.1345" velocity="2.0944"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <joint name="joint_4" type="revolute">
    <origin rpy="0 0 0" xyz="0.63 0 0.15"/>
    <parent link="link_3"/>
    <child link="link_4n"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.49" upper="3.49" velocity="3.927"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- .............................................................................. -->
  <!-- joinit new -->
  <joint name="joint_4n" type="prismatic">
    <parent link="link_4n"/>
    <child link="link_4"/>
    <origin rpy="0 0 0" xyz="1.10 0 0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-0.35" upper="0.35" velocity="0.1"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <!-- joint new end -->
  <!-- .............................................................................. -->
  <joint name="joint_5" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_4"/>
    <child link="link_5"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-2.0944" upper="2.0944" velocity="4.3633"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <joint name="joint_6" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_5"/>
    <child link="link_6"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-6.9813" upper="6.9813" velocity="5.7596"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <joint name="joint_6-tool0" type="fixed">
    <parent link="link_6"/>
    <child link="tool0"/>
    <origin rpy="0 0 0" xyz="0.14 0 0"/>
  </joint>
  <!-- end of joint list -->

 
  <!--Transmisions-->

  <transmission name="joint1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!--<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>-->
    </joint>
    <actuator name="joint1_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="joint2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!--<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>-->
    </joint>
    <actuator name="joint2_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>


  <transmission name="joint3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!--<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>-->
    </joint>
    <actuator name="joint3_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>


  <transmission name="joint4_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!--<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>-->
    </joint>
    <actuator name="joint4_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="joint4n_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_4n">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!--<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>-->
    </joint>
    <actuator name="joint4n_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="joint5_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!--<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>-->
    </joint>
    <actuator name="joint5_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="joint6_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!--<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>-->
    </joint>
    <actuator name="joint6_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <!-- <link name="world"/> -->
  <link name="world">
    <inertial>
      <mass value="0.0000001"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0000001" ixy="0.0" ixz="0.0" iyy="0.0000001" iyz="0.0" izz="0.0000001"/>
    </inertial>
  </link>
    
  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>

   <!-- Plugin: ros_control -->
   <gazebo>
     <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
       <robotNamespace>/robot</robotNamespace>
       <legacyModeNS>true</legacyModeNS>
     </plugin>
   </gazebo>
</robot>

